staging/src/k8s.io/apiextensions-apiserver/pkg/apiserver/customresource_handler.go

```diff
@@ -940,6 +1052,33 @@ func (r *crdHandler) getOrCreateServingInfoFor(uid types.UID, name string) (*crd
 	storageMap2[crd.UID] = ret
 	r.customStorage.Store(storageMap2)
 
+	if !utilfeature.DefaultFeatureGate.Enabled(features.StorageVersionAPI) ||
+		!utilfeature.DefaultFeatureGate.Enabled(features.APIServerIdentity) {
+		return ret, nil
+	}
+
+	q := r.getOrCreateStorageVersionUpdateQueue_locked(crd.UID)
+	if len(q) == maxStorageVersionUpdateRequests {
+		// When the channel buffer is full, writing to the channel becomes blocking.
+		// Since we still hold the customStorageLock when we write to the channel,
+		// a too-frequently-updated CRD can block the entire API server. Here we
+		// give up updating storage version for this CRD and print a log.
+		// TODO(roycaihw): use warning instead of info when StorageVersionAPI graduates to beta/GA
+		klog.V(4).Infof("Skipping the storage version update for CRD with UID %v due to the queue being full (queue size: %v).",
```

Comment:
This should be a V(2). This is expected to be rarely triggered, and when it's triggered, we do want to see it in the logs.

Reply:
Done.